cmd/metal-api/internal/metal/provisioning.go

```diff
@@ -41,6 +42,7 @@ var (
 		ProvisioningEventInstalling:       true,
 		ProvisioningEventBootingNewKernel: true,
 		ProvisioningEventPhonedHome:       true,
+		ProvisioningEventReinstallAborted: true,
```

Comment:
This will actually indicate for the state machine that this is the "final state" of machine provisioning. The reinstall should not be part of the regular provisioning sequence. Instead, it should go into the `expectedSuccessorEventMap` where you can define from which state it is allowed to transition into reinstall and to which state it is allowed to proceed after the reinstall.

It would also be nice to have a test for that. You can just use example sequences of provisioning events and check whether the state machine behaves as expected or not.